### DISCUSSION TITLE
Emails non délivrés

### DISCUSSION THREAD

**Discussion by alexisig** [MTES-MCT/sparte#848] [Idées]
Situation : 
3 cas les plus fréquents (classé par fréquence) :
1) Adresse email mal renseignée par l'utilisateur (90%)
2) Email bloqué par un service anti-spam ou un anti-virus (5%)
3) Serveur d'email de l'utilisateur temporairement inaccessible (5%)

-----

Pour répondre au cas n°1 :
A) Rendre le téléchargement des diagnostics limité aux utilisateurs authentifiés -> pas pour l'instant
B) Ajouter une étape de validation d'adresse email qui conditionne la possibilité de télécharger un diag.
-> préciser de vérifier dans la boite spam
C) Si email confirmé : ne plus afficher de formulaire de téléchargement, mais télécharger directement

Pour répondre au cas n°2 : 
C) Indiquer à l'utilisateur anonyme une fois le formulaire de téléchargement rempli de penser à vérifier son dossier 'SPAM'
D) Indiquer dans la documentation que les mails non delivrés peuvent venir de logiciel anti-spam (ex ALTOSPAM) ou anti-virus (ex Sophos)

Pour répondre au cas n°3 : 
E) Aujourd'hui nous réessayons plusieurs fois de renvoyer un même email, mais sur une période de temps courte. Parfois plus courte qu'il ne faut de temps au serveur destinataire pour être de nouveau actif. Pour compenser ce problème : réessayer une dernière au minimum 24h après le premier essai.

**Comment by alexisig** [comment]
@JeremieSeguin est-ce que tu vois d'autres choses ?

**Comment by JeremieSeguin** [comment]
> @JeremieSeguin est-ce que tu vois d'autres choses ?

Pas spécialement d'autres choses, une idée qui allègerait le parcours utilisateur : 

- En appliquant la réponse au cas n°1, on pourrait supprimer le formulaire de téléchargement (puisque les utilisateurs sont connectés, on devrait déjà détenir les infos contenues dans le formulaire. 
- Cela impliquerait de transformer le C) en simple pop up avec un bouton "j'ai compris" par exemple ?

**Comment by alexisig** [comment]
> > @JeremieSeguin est-ce que tu vois d'autres choses ?
> 
> Pas spécialement d'autres choses, une idée qui allègerait le parcours utilisateur :
> 
> * En appliquant la réponse au cas n°1, on pourrait supprimer le formulaire de téléchargement (puisque les utilisateurs sont connectés, on devrait déjà détenir les infos contenues dans le formulaire.
> * Cela impliquerait de transformer le C) en simple pop up avec un bouton "j'ai compris" par exemple ?

Merci de ton retour ! Très bonne idée, j'édite le corps du premier message

**Comment by leabouche** [comment]
A découper 
- validation email 
- telechargement direct pour les users connectés

**Comment by leabouche** [comment]
lié à https://github.com/MTES-MCT/sparte/issues/585
